### PR TITLE
ft: add and update ratings

### PR DESCRIPTION
**Pull request by blackd44**
# PR Description

We have to make sure that the coordinator can add a review and can update it.

# Description of tasks that were expected to be completed

- [x] coordinator add ratings to trainees
- [x] coordinator add feedback on rate
- [x] trainee add feedback on rate
- [x] update ratings
- [x] admin approve ratings updates

# How has this been tested?
* For unity nesting, run `npm test` command to test
* For verifying the functionality
   ```bash
    git clone https://github.com/atlp-rwanda/atlp-pulse-fn
    cd atlp-pulse-fn
    ```
    go to the branch called `ft-add-ratings` then
    ```bash
     npm install
     npm run dev
     open http://localhost:3000
    ```
    - ### for adding ratings
        - login as a coordinator with these credentials:  
             `email: georgia.gonzales@devpulse.com`
              `Password : Test@12345`
        - in side bar, click on Ratings link to reach ratings
        - click on `add new ratings` button
        - add required data and continue
    - ### adding feedback to rating remarks
        - login as a coordinator
        - 
           * for `coordinator` go to ratings in sidebar, you will find `chat icon` button in the ratings table
           * for `trainee` go to performance in sidebar, i you will find `details` button in the ratings table
        - then click it to open model
        - the add your feedback
    - ### Updating rating
        - login as a coordinator
        - go to ratings in sidebar, you will find `write icon` button in the ratings table
        - then click it to open model
        - the add your updates then submit
    - ### approve and reject updates
        - login as a admin
        - go to `Updated Ratings` in sidebar
        - You will find 2 buttons in the table
          * click ✅ button approve
          * click ❌ to reject 
        - those button will open models and click the `approve` or `reject` buttons to continue
# Number of Commits

**1 commit**

# Screenshots (If appropriate)
Screenshot 2023-06-30 at 11 01 59
Screenshot 2023-06-30 at 11 02 54
Screenshot 2023-06-30 at 11 17 38
Screenshot 2023-07-12 at 17 23 53

# Please check this Checklist before you submit your PR:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code generate no warnings
- [x] My test coverage meet the set test coverage threshold
- [x] There are no vulnerabilities
- [x] There are no conflicts with the base branch

### CODE REVIEW

**Review by UwaweTera — APPROVED** [review]

**Review by Patienceineza — APPROVED** [review]
LGTM

**Review by edwinnambaje — APPROVED** [review]

**Review by ceelogre — CHANGES_REQUESTED** [review]
When we type a feedback message, it does not appear in the box. Also, let's add a title that shows who is the trainee we are giving feedback.
image

A message that a rating has been updated is shown while the admin hasn't approved it. A better message like "waiting for admin approval" would be suited.
image

**Comment by blackd44** [comment]
> When we type a feedback message, it does not appear in the box. Also, let's add a title that shows who is the trainee we are giving feedback. image
> 
> A message that a rating has been updated is shown while the admin hasn't approved it. A better message like "waiting for admin approval" would be suited. image

the reason why the feedback wasn't working it's because there wasn't environment variable of `WS_BACKEND_URL`

**Review by ceelogre — APPROVED** [review]

**Pull request merged**
